source/common/access_log/access_log_impl.h

```diff
@@ -166,6 +166,21 @@ class HeaderFilter : public Filter {
   std::vector<Http::HeaderUtility::HeaderData> header_data_;
 };
 
+/**
+ * Filter requests that had a response with an Envoy response flag set.
+ */
+class ResponseFlagFilter : public Filter {
+public:
+  ResponseFlagFilter(const envoy::config::filter::accesslog::v2::ResponseFlagFilter& config);
+
+  // AccessLog::Filter
+  bool evaluate(const RequestInfo::RequestInfo& info,
+                const Http::HeaderMap& request_headers) override;
+
+private:
+  RequestInfo::RequestInfoImpl info_;
```

Comment:
nit: It seems a little odd to me to have a full request info here, but it's probably fine. Just throwing that out there.

Reply:
Yeah, it is a little clunky. I guess I could just maintain the `uint64_t` for the flags and instead of using the setResponseFlag abstraction do the `|=` myself. 

That way I also can get rid of the `currentResponseFlags()` method and add less stuff to that api.

wdyt?

Reply:
yeah I would probably do that...